src/Map/src/Renderer/Renderers.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map\Renderer;
+
+use Symfony\UX\Map\Exception\LogicException;
+use Symfony\UX\Map\Map;
+
+/**
+ * @author Hugo Alliaume <hugo@alliau.me>
+ */
+class Renderers implements RendererInterface
+{
+    /**
+     * @var array<string, RendererInterface>
+     */
+    private array $renderers = [];
+    private RendererInterface $default;
+
+    /**
+     * @param iterable<string, RendererInterface> $renderers
+     */
+    public function __construct(iterable $renderers)
+    {
+        foreach ($renderers as $name => $renderer) {
+            $this->default ??= $renderer;
+            $this->renderers[$name] = $renderer;
+        }
+
+        if (!$this->renderers) {
+            throw new LogicException(\sprintf('"%s" must have at least one renderer configured.', __CLASS__));
+        }
+    }
+
+    public function renderMap(Map $map, array $attributes = []): string
+    {
+        $renderer = $this->renderers[$map->getRendererName()] ?? $this->default;
```

Comment:
If rendererName is set and is not available, not  sure we should return the default one

Reply:
You're right, let's throw if necesary